### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by MihaiTabara**
CC @jmaher @sarah-clements @jcristau

### CODE REVIEW

File: procedures/release-duty/merge-duty/merge_duty.rst [hunk 1]
```diff
@@ -24,7 +24,6 @@ nearly three weeks, with *three* major days of activity:
 -  Do the prep work a week before the merge
 
    -  `File tracking migration bug <#file-tracking-migration-bug>`__
-   -  `Turn off beta l10n bumper on RC day <#turn-off-beta-l10n-bumper-on-rc-day>`__
```
Comment (MihaiTabara) [inline procedures/release-duty/merge-duty/merge_duty.rst:27]: Courtesy to @jcrisau's work in https://bugzil.la/1680028, l10n bumper runs on a daily basis now. Given that, I don't think it's worth doing this operation, which at this point would only save spinning up a no-op (given beta is closed) worker 7 times until next mergeduty Monday.

File: procedures/release-duty/merge-duty/merge_duty.rst [hunk 2]
```diff
@@ -137,9 +129,9 @@ Make sure the bug that tracks the migration has no blocking items.
 Land whatsnewpage list of locales
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-**TODO** - this needs to change, as the process no longer assumes this,
```
Comment (MihaiTabara) [inline procedures/release-duty/merge-duty/merge_duty.rst:140]: @jcristau do you think we should remove this section altogether or keep it for now until we have a few successful releases ironing out the new WNP process?

Comment (jcristau) [reply procedures/release-duty/merge-duty/merge_duty.rst:140]: I think we're probably OK removing it from the regular merge duty doc given https://bugzilla.mozilla.org/show_bug.cgi?id=1680091#c0

**Review by jmaher — APPROVED** [review]
glad to see more cleanup!

**Pull request closed without merging**